manifests/puppetserver/autoload.pp

```diff
@@ -1,14 +1,13 @@
 class ducktape::puppetserver::autoload() {
 
-  $defaults = {'notify' => Service['puppetserver']}
-  create_resources('ducktape::puppetserver::autoload::config::puppetserver', hiera_hash('ducktape::puppetserver::config::puppetserver', {}), $defaults)
+  create_resources('ducktape::puppetserver::autoload::config::puppetserver', $ducktape::puppetserver::puppetserver_config, $ducktape::puppetserver::defaults)
 
 }
 
 define ducktape::puppetserver::autoload::config::puppetserver(
-  String $file = 'puppetserver.conf',
-  $value,
-  $setting_type = undef,
+  String                                   $file         = 'puppetserver.conf',
+  Optional[Variant[Array[String], String]] $value        = undef,
```

Comment:
The parameter `value` had no default value, so for consistency I have added `undef` as its default value.

Additionally to decide the type:
1. Since it is undefined it must be `Optional` or else puppet execution could fail if no value is provided
2. Watching our code in core, this parameter will recieve either a String or an array of strings:
```yaml
ducktape::puppetserver::config::puppetserver:
  server-name1:
    value: ['/opt/puppetlabs/puppet/lib/ruby/XXXX', '/opt/puppetlabs/puppet/cache/lib']
    setting_type: array
  server-name2:
    value: 'force'
```

Keeping all the possible values of this parameter in mind I believe that this is the correct way to change it without having to make changes in core.

Reply:
Si una variable no tiene valor por defecto, es que es requerida. Si le pones = undef deja de ser requerida. Esta decisión incorrecta lleva a su vez a meter Optional para que puppet no se queje.

Variant ok, pero quita el Optional y el undef.

Reply:
Done